optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/tri/DroolsTriCondition.java

```diff
@@ -123,7 +118,15 @@ public DroolsTriCondition(DroolsMetadata<Object, A> aMetadata, DroolsMetadata<Ob
                 on(scoreHolderGlobal, aMetadata.getVariableDeclaration(), bMetadata.getVariableDeclaration(),
                         cMetadata.getVariableDeclaration())
                         .execute(consequenceImpl);
-        return Arrays.asList(aMetadata.getPattern(), bMetadata.getPattern(), cMetadata.getPattern(), consequence);
+        if (aMetadata instanceof DroolsInferredMetadata && bMetadata instanceof DroolsInferredMetadata &&
+                cMetadata instanceof DroolsInferredMetadata) {
+            // In case of logical tuples, all patterns will be the same logical tuple, and therefore we just add one.
+            return Stream.of(cMetadata.buildPattern(), consequence)
+                    .collect(Collectors.toList());
+        } else {
+            return Stream.of(aMetadata.buildPattern(), bMetadata.buildPattern(), cMetadata.buildPattern(), consequence)
+                    .collect(Collectors.toList());
```

Comment:
I suspect that `Arrays.asList(aMetadata.buildPattern(), bMetadata.buildPattern(), cMetadata.buildPattern(), consequence)` is faster and less verbose?

Reply:
Yeah. This is a leftover from a time when this code was doing something more complex. I'll change it.